outline/tun2socks/README.md

```diff
@@ -0,0 +1,94 @@
+# outline-apps/outline/tun2socks
+
+Go package for building [go-tun2socks](https://github.com/eycorsican/go-tun2socks)-based clients for [Outline](https://getoutline.org) and [Intra](https://getintra.org) (now with support for [Choir](https://github.com/Jigsaw-Code/choir) metrics). For macOS, iOS, and Android, the output is a library; for Linux and Windows it is a command-line executable.
+
+## Prerequisites
+
+- macOS host (iOS, macOS)
+- make
+- Go >= 1.18
+- A C compiler (e.g.: clang, gcc)
+
+## Android
+
+### Set up
+
+- [sdkmanager](https://developer.android.com/studio/command-line/sdkmanager)
+  1. Download the command line tools from https://developer.android.com/studio.
+  1. Unzip the pacakge as `~/Android/Sdk/cmdline-tools/latest/`. Make sure `sdkmanager` is located at `~/Android/Sdk/cmdline-tools/latest/bin/sdkmanager`
+- Android NDK 23
+  1. Install the NDK with `~/Android/Sdk/cmdline-tools/latest/bin/sdkmanager "platforms;android-30" "ndk;23.1.7779620"` (platform from [outline-client](https://github.com/Jigsaw-Code/outline-client#building-the-android-app), exact NDK 23 version obtained from `sdkmanager --list`)
+  1. Set up the environment variables:
+     ```
+     export ANDROID_NDK_HOME=~/Android/Sdk/ndk/23.1.7779620 ANDROID_HOME=~/Android/Sdk
+     ```
+- [gomobile](https://pkg.go.dev/golang.org/x/mobile/cmd/gobind) (installed as needed by `make`)
+
+### Build
+
+```bash
+make clean && make android
```

Comment:
Not sure whether we still need this, because I think the process will be embedded into `npm build` when building the client.